### ISSUE TITLE
[ Feature Request ] Custom Keybindings for various actions + vi mode

### ISSUE THREAD

**Issue by ghost** [ScratchAddons/ScratchAddons#3902]
### Add keybindings for the following actions:
 - Cleanup +
 - Add Comment
 - Scroll {Up, Down, Left, Right}
 - Drag Block {Up, Down, Left, Right}
 - Switch Editor Tabs
 - Switch Sprites
 - Switch to the Backdrop
 - {Start, Pause, Stop, Restart} the Project
 - Show the debugger console
 - Create new Custom Block
 - Create new variable
 - Create new list

### Add the ability to load custom keybinds from a configuration file:
The syntax for the config file could be like this:
```
ctrl + S = save
ctrl + shift + R = restart_project
```

### Add keybindings from the Vi editor

**Comment by WorldLanguages** [comment]
The vi mode will need a separate issue with more details.
But adding hotkeys for common actions is something we are interested in. The settings page could have a hotkeys section that not only let you customize keybinds used by addons, but also the ones already existing on Scratch (e.g. Ctrl+Z) as well as those that don't exist (keybind to switch tabs or create a new variable)

**Comment by TheColaber** [comment]
Part of the reason I haven't already pr'ed for this, was because Hotkey combinations in websites are somewhat limited. Most of the Ctrl Combos are taken unless you require the Shift key to also be held down. Alt is still an option however. Or maybe sometimes, A key like Ctrl or Alt don't need to be pressed, and It could be as easy as a single letter or number. No matter what we do, the `key` setting type needs to be changed to implement Alt, or alphabet and numeric keys. A new API should also be written. Blockly creates a listener for the editor, which we  should may or may not use for editor hotkeys. I'm not sure how the paint editor with paper and redux works, but maybe somebody knows and can put that here. The last option is to use document.addEventListener.

**Comment by RedGuy12** [comment]
Well we could make the keybinds userconfigurable and have them override Scratch/browser defaults


________________________________

**Comment by TheColaber** [comment]
Some you can't override, like Ctrl + T or Ctrl + H.

**Comment by WorldLanguages** [comment]
Yeah, it will be complicated to choose the default keybinds, since browsers and OSes take many of them

**Comment by TheColaber** [comment]
> Yeah, it will be complicated to choose the default keybinds, since browsers and OSes take many of them

Not only that, but also making sure that the user doesn't select one of these used keybinds.
